README.md

```diff
@@ -1,5 +1,15 @@
 # Unified Memory Framework
 
+The Unified Memory Framework (UMF) is a library for constructing allocators and memory pools. It also contains broadly useful abstractions and utilities for memory management. UMF allows users to manage multiple memory pools characterized by different attributes, allowing certain allocation types to be isolated from others and allocated using different hardware resources as required.
+
+# Architecture: memory pools and providers
+
+A UMF memory pool is a combination of a pool allocator and a memory provider. A memory provider is responsible for coarse-grained memory allocations and management of memory pages, while the pool allocator controls memory pooling and handles fine-grained memory allocations.
+
+Pool allocator can leverage existing allocators (e.g. jemalloc or tbbmalloc) or be written from scratch. 
+
+UMF comes with a variety of pool allocators (see include/pool) and providers (see include/provider). UMF can also work with user-defined pools and providers that implement a specific interface (see include/umf/memory_pool_ops.h and include/umf/memory_provider_ops.h)
```

Comment:
Since we have no "vasriety" yet :) maube better to say "UMF comes with predefined pool allocators (...) and providers (...)"

Reply:
Yeah, that was more of a forward-looking statement ;d Done.